lib/doorkeeper/oauth/client_credentials/creator.rb

```diff
@@ -5,11 +5,25 @@ module OAuth
     class ClientCredentialsRequest < BaseRequest
       class Creator
         def call(client, scopes, attributes = {})
+          existing_token = existing_token_for(client, scopes)
+
+          if Doorkeeper.configuration.reuse_access_token && existing_token&.reusable?
+            return existing_token
+          end
+
+          existing_token&.revoke
```

Comment:
Remind me please why we are revoking the token ?

Reply:
The `authorized_tokens_for` scope assumes that old tokens will be revoked: https://github.com/doorkeeper-gem/doorkeeper/blob/master/lib/doorkeeper/models/access_token_mixin.rb#L156

Making this change causes client credentials to be performant in this case.  Since we only revoke the token when the previous one is either expired or can't be reused, we're not actually revoking a token that could otherwise be used.

Reply:
For reference, I created this ticket in response to this comment: https://github.com/doorkeeper-gem/doorkeeper/issues/1314

Reply:
Some fixes for those who will find this topic: #1353 (Doorkeeper 5.3)